pkg/image/resolve_test.go

```diff
@@ -0,0 +1,369 @@
+	}{
+		{
+			tName:   "linux",
+			iPath:   systemChartPath,
+			iOsType: Linux,
+			oShouldContain: []string{
+				"rancher/prom-alertmanager:v0.17.0",
+				"rancher/configmap-reload:v0.3.0-rancher2", // both windows and linux
+				"rancher/fluentd:v0.1.16",                  // both windows and linux
+			},
+			oShouldNotContain: []string{
+				"rancher/wmi_exporter-package:v0.0.2",
+			},
+		},
+		{
+			tName:   "windows",
+			iPath:   systemChartPath,
+			iOsType: Windows,
+			oShouldContain: []string{
+				"rancher/wmi_exporter-package:v0.0.2",
+				"rancher/configmap-reload:v0.3.0-rancher2", // both windows and linux
+				"rancher/fluentd:v0.1.16",                  // both windows and linux
+			},
+			oShouldNotContain: []string{
+				"rancher/prom-alertmanager:v0.17.0",
+			},
+		},
+	}
+
+	assert := assertlib.New(t)
+	for _, cs := range testCases {
+		images, err := fetchImagesFromCharts(cs.iPath, cs.iOsType)
+		assert.Nilf(err, "%s, failed to fetch images from charts", cs.tName)
+
+		if len(cs.oShouldContain) > 0 {
+			assert.Subset(images, cs.oShouldContain, cs.tName)
+		}
+		if len(cs.oShouldNotContain) > 0 {
+			for _, nc := range cs.oShouldNotContain {
+				assert.NotContains(images, nc, cs.tName)
+			}
+		}
+	}
+}
+
+func TestFetchImagesFromSystem(t *testing.T) {
+	linuxInfo, windowsInfo := getK8sVersionInfo()
+	toolsSystemImages := v3.ToolsSystemImages
+	rkeSystemImages := getRKESystemImages()
+
+	testCases := []struct {
+		tName             string
+		iRkeSystemImages  map[string]v3.RKESystemImages
+		iOsType           OSType
+		oShouldContain    []string
+		oShouldNotContain []string
+	}{
+		{
+			tName:            "linux",
+			iRkeSystemImages: linuxInfo.RKESystemImages,
+			iOsType:          Linux,
+			oShouldContain: []string{
+				toolsSystemImages.PipelineSystemImages.Jenkins,
+				toolsSystemImages.PipelineSystemImages.JenkinsJnlp,
+			},
+			oShouldNotContain: []string{
+				rkeSystemImages.WindowsPodInfraContainer,
+			},
+		},
+		{
+			tName:            "windows",
+			iRkeSystemImages: windowsInfo.RKESystemImages,
+			iOsType:          Windows,
+			oShouldContain: []string{
+				rkeSystemImages.WindowsPodInfraContainer,
+				rkeSystemImages.NginxProxy,
+			},
+			oShouldNotContain: []string{
+				toolsSystemImages.PipelineSystemImages.Jenkins,
+				toolsSystemImages.PipelineSystemImages.JenkinsJnlp,
+			},
+		},
+	}
+
+	assert := assertlib.New(t)
+	for _, cs := range testCases {
+		images, err := fetchImagesFromSystem(cs.iRkeSystemImages, cs.iOsType)
+		assert.Nilf(err, "%s, failed to fetch images from system images", cs.tName)
+
+		if len(cs.oShouldContain) > 0 {
+			assert.Subset(images, cs.oShouldContain, cs.tName)
+		}
+		if len(cs.oShouldNotContain) > 0 {
+			for _, nc := range cs.oShouldNotContain {
+				assert.NotContains(images, nc, cs.tName)
+			}
+		}
+	}
+}
+
+func TestAppendImagesFromRequirement(t *testing.T) {
+	testCases := []struct {
+		tName             string
+		iImages           []string
```

Comment:
You don't have any test cases including `iImages` with content. Is that expected?

Reply:
Using `getRequirementImages` instead of `appendImagesFromRequirement`, so remove this case.